src/Validator/InputValidator.php

```diff
@@ -154,18 +167,29 @@ private function buildValidationTree(ValidationNode $rootObject, iterable $field
                 }
 
                 $valid = new Valid();
+                $groups = $config['cascade'];
 
                 if (!empty($groups)) {
                     $valid->groups = $groups;
                 }
 
+                // Apply the Assert/Valid constraint for a recursive validation.
+                // For more details see https://symfony.com/doc/current/reference/constraints/Valid.html
                 $metadata->addPropertyConstraint($property, $valid);
+
+                // Skip the rest as the validation was delegated to the nested object.
+                continue;
```

Comment:
There is no need to proceed and add any more constraints if it was set to `cascade`